src/Controls/src/SourceGen/CodeBehindGenerator.cs

```diff
@@ -463,33 +465,75 @@ static void GenerateCssCodeBehind(ProjectItem projItem, SourceProductionContext
 			sourceProductionContext.AddSource(hintName, SourceText.From(sb.ToString(), Encoding.UTF8));
 		}
 
-		static void ApplyTransforms(XmlNode node)
+		static void ApplyTransforms(XmlNode node, string? targetFramework, XmlNamespaceManager nsmgr)
 		{
-			//if (release)
-			//	SimplifyOnPlatform(node, targetFramework);
+				SimplifyOnPlatform(node, targetFramework, nsmgr);
 		}
 
-		static void SimplifyOnPlatform(XmlNode node, string targetFramework)
+		static void SimplifyOnPlatform(XmlNode node, string? targetFramework, XmlNamespaceManager nsmgr)
 		{
+			//remove OnPlatform nodes if the platform doesn't match, so we don't generate field for x:Name of elements being removed
+			if (targetFramework == null)
+				return;			
+
+			string? target = null;
+			targetFramework = targetFramework.Trim();
+			if (targetFramework.Contains("-android"))
+				target = "Android";
+			if (targetFramework.Contains("-ios"))
+				target = "iOS";
+			if (targetFramework.Contains("-macos"))
+				target = "macOS";
+			if (targetFramework.Contains("-maccatalyst"))
+				target = "MacCatalyst";
```

Comment:
A customer could actually put `net7.0-Android`... And things still work fine.

Should we use `StringComparison.OrdinalIgnoreCase` for all of these cases?

Reply:
I don't think we have StringComparison options in netstandard2.0, and afaik, analysers should be ns2.0